test/integration/scheduler/filters/filters_test.go

```diff
@@ -2036,6 +2037,53 @@ func TestUnschedulablePodBecomesSchedulable(t *testing.T) {
 				return deletePod(cs, "pod-to-be-deleted", ns)
 			},
 		},
+		{
+			name: "pod with pvc has node-affinity to non-existent/illegal nodes",
```

Comment:
Can we also add a case which pod got scheduled after node updated its label?

Maybe sth like this:

1. Pod with pv's affinity with `foo in bar` created
2. Node with label `foo = foo` created
3. Verify that the pod is not scheduled
4. Update the node with label `foo = bar`
5. Verify that the pod is scheduled successfully.